meson.build

```
@@ -0,0 +1,519 @@
+  conf.set('PCI_HAVE_PM_WIN32_CFGMGR32', 1)
+  conf.set('PCI_HAVE_PM_WIN32_KLDBG', 1)
+  conf.set('PCI_HAVE_PM_WIN32_SYSDBG', 1)
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += [
+      'lib/emulated.c',
+      'lib/win32-cfgmgr32.c',
+      'lib/win32-kldbg.c',
+      'lib/win32-sysdbg.c',
+    ]
+    + i386_port_srcs
+  endif
+  dependencies += [ cc.find_library('ioperm', required: true) ]
+elif host_machine.system() == 'windows'
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  conf.set('PCI_HAVE_PM_WIN32_CFGMGR32', 1)
+  conf.set('PCI_HAVE_PM_WIN32_KLDBG', 1)
+  conf.set('PCI_HAVE_PM_WIN32_SYSDBG', 1)
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += [
+      'lib/emulated.c',
+      'lib/win32-cfgmgr32.c',
+      'lib/win32-kldbg.c',
+      'lib/win32-sysdbg.c',
+    ]
+    + i386_ports_srcs
+  endif
+  dependencies += [ cc.find_library('advapi32', required: true) ]
+elif host_machine.system() == 'haiku'
+  if host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64'
+    conf.set('PCI_HAVE_PM_INTEL_CONF', 1)
+    sources += i386_ports_srcs
+  endif
+  conf.set('PCI_HAVE_STDINT_H', 1)
+elif host_machine.system() == 'sylixos'
+  conf.set_quoted('PCI_PATH_SYLIXOS_DEVICE', '/proc/pci')
+  conf.set('PCI_HAVE_64BIT_ADDRESS', 1)
+  conf.set('PCI_HAVE_PM_SYLIXOS_DEVICE', 1)
+  IDSDIR = get_option('prefix') / get_option('sysconfdir') / 'pci'
+  sources += ['lib/sylixos-device.c']
+else
+  error('Unfortunately, your OS is not supported by the PCI library')
+endif
+
+conf.set_quoted('PCI_PATH_IDS_DIR', IDSDIR)
+
+conf.set10('PCI_COMPRESSED_IDS', zlib_dep.found())
+if zlib_dep.found()
+  conf.set_quoted('PCI_IDS', 'pci.ids.gz')
+  pci_ids = 'pci.ids.gz'
+else
+  conf.set_quoted('PCI_IDS', 'pci.ids')
+  pci_ids = 'pci.ids'
+endif
+
+if (get_option('with-dns').enabled() or
+  get_option('with-dns').auto() and host_machine.system() != 'windows')
+
+  assert(cc.has_function('res_query', dependencies:dns_dep))
+  conf.set('PCI_USE_DNS', 1)
+  conf.set_quoted('PCI_ID_DOMAIN', 'pci.id.ucw.cz')
+endif
+
+conf.set10('PCI_USE_LIBKMOD', kmod_dep.found())
+conf.set10('PCI_HAVE_HWDB', hwdb_dep.found())
+conf.set10('PCI_HAVE_PM_DUMP', true)
+sources += ['lib/dump.c']
+
+conf_file = configure_file(configuration: conf, output: 'config.h')
+
+pcilib = library('pci',
+                 dependencies: dependencies,
+                 install: true,
+                 sources: sources,
+                 gnu_symbol_visibility: 'hidden',
+                 link_args: '-Wl,--version-script=' + '../lib/libpci.ver',
+                 link_depends: ['lib/libpci.ver'])
+
+pkg = import('pkgconfig')
+pkg.generate(pcilib,
+             name: 'libpci',
+             filebase: 'libpci',
+             description:'libpci',
+             variables: {'idsdir': IDSDIR})
```

This can be changed to:
```suggestion
             variables: ['idsdir= +'IDSDIR])
```

further dropping the minimum meson version to 0.53.0